text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+metadata like PURL and CPEs.
+
+Libraries like `libpak` and `packit` then provide convenience methods for
+fetching dependencies using this metadata, verifying the download, and caching
+the download result. In addition, they provide tooling to download and store
+these dependencies within buildpack images for distribution in offline
+environments.
+
+There are also tools published by the project to manage the entries within
+`buildpack.toml` through CI pipelines so that dependencies metadata is kept
+up-to-date with upstream sources. Unfortunately, this represents a large amount
+of toil for the buildpacks team.
+
+As an example of the toil mentioned in a language family like Java, there are
+daily project dependencies that need to be updated. This requires reviewing and
+merging PRs into the buildpacks to adjust `buildpack.toml` dependency metadata.
+Once PRs are merged, a component buildpack needs to be released, followed by a
+composite buildpack and then a builder release. This is because most users
+don’t consume buildpacks directly, they consume builders which include
+buildpacks.
+
+This all has to be done as aggressively as possible so that we are shipping
+dependencies, in particular those with security fixes, quickly. This is because
+with metadata in `buildpack.toml`, even if an upstream project releases a bug
+or security fix, buildpack users cannot get that fix until we update and
+release component and composite buildpacks as well as the builder.
+
+There is also toil associated with the tools and pipelines used for this
+process. The tools have bugs and need to be updated. At present, the tools we
+use to manage all of these updates do not scale well either. In particular
+Github Actions, we have had a number of issues hitting rate limits and usage
+caps. This gets worse when there are a lot of dependencies to watch, for
+example, if your buildpack has multiple version lines or different sets of
+packages for a dependency.
+
+The whole process puts an additional maintenance burden on the project
+maintainers and project resources. This is not the type of work that a casual
+contributor to Paketo will do and as we add more dependencies the burden only
+increases on the maintainer teams.
+
+The motivation of this proposal is to…
+
+- Reduce the burden and toil for Paketo buildpack maintainer teams
+- Continue publishing dependency updates in a timely and secure manner
+- Decouple installing dependencies from configuring them
+- Separate metadata and the actual dependencies so they can be provided to
+  buildpacks in a number of different and flexible ways
+- Establish a reasonable release schedule for buildpacks that’s based around
+  development, not dependencies and thus enabling buildpacks to support version
+  lines
+- Make it easier to package buildpacks for offline environments.
+
+## Detailed Explanation
+At a high level:
+
+- We will define a metadata format that includes the metadata fields currently
```

Comment:
I think I'd add two bullet points (probably second and third in the list) that say something like:

* This newly-defined metadata format replaces the existing dependency metadata in `buildpack.toml`
* This metadata is provided at build-time in a known location on the build-time filesystem

Reply:
Funny, I guess I've subconsciously just cited it from here ;) I might give the whole paragraph a closer look and see how we can best arrange it.